### PR TITLE
Fix a problem reported by a linter by recoding if statement through a…

### LINKED ISSUE #3555: make testall fails because of a problem reported by linter
## Version of Singularity:

Master on Ubuntu.

### Expected behavior

make checkall should succeed.

### Actual behavior

```
$ make testall -C builddir/
make: Entering directory '/home/gvallee/go/src/github.com/sylabs/singularity/builddir'
build flag -mod=vendor only valid when using modules
build flag -mod=vendor only valid when using modules
 CHECK golangci-lint
internal/pkg/runtime/engines/singularity/process_linux.go:312:40: nilness: tautological condition: nil == nil (govet)
					if wpid == cmd.Process.Pid && err == nil {
					                                  ^
make: *** [Makefile:427: check] Error 1
make: Leaving directory '/home/gvallee/go/src/github.com/sylabs/singularity/builddir'
```

### Steps to reproduce behavior

`$ make testall -C builddir/`

### PR DESCRIPTION
**Pull request by gvallee**
… switch statement

**Description of the Pull Request (PR):**

Make the linter happy.

**This fixes or addresses the following GitHub issues:**

- Fixes #3555 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

**Review by mem — APPROVED** [review]

File: internal/pkg/runtime/engines/singularity/process_linux.go [hunk 1]
```diff
@@ -301,14 +301,19 @@ func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
...
 			switch s {
 			case syscall.SIGCHLD:
+			InnerLoop:
 				for {
 					var status syscall.WaitStatus
 
 					wpid, err := syscall.Wait4(-1, &status, syscall.WNOHANG, nil)
-					if wpid <= 0 || err != nil {
-						break
-					}
```
Comment (cclerget) [inline internal/pkg/runtime/engines/singularity/process_linux.go]: Would linter be happy with `} else if wpid == cmd.Process.Pid && err == nil {` instead ?

**Comment by gvallee** [comment]
Unfortunately no, it still does not like the err == nil.

**Review by cclerget — CHANGES_REQUESTED** [review]
> Unfortunately no, it still does not like the err == nil.

Well I don't know why I put an extra `err == nil` while writing those checks, because `wpid > 0` should imply `err == nil`, so it should be safe to just write `} else if wpid == cmd.Process.Pid {` to make linter happy.

**Comment by gvallee** [comment]
Updated

File: internal/pkg/runtime/engines/singularity/process_linux.go [hunk 2]
```diff
@@ -306,9 +306,15 @@ func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
 
 					wpid, err := syscall.Wait4(-1, &status, syscall.WNOHANG, nil)
 					if wpid <= 0 || err != nil {
+						// We break the loop since an error occurred
 						break
 					}
-					if wpid == cmd.Process.Pid && err == nil {
+
+					if wpid <= 0 {
```
Comment (cclerget) [inline internal/pkg/runtime/engines/singularity/process_linux.go]: this condition will be never met, you can remove it

**Review by cclerget — CHANGES_REQUESTED** [review]

File: internal/pkg/runtime/engines/singularity/process_linux.go (hunk 2 above)
Comment (gvallee) [reply internal/pkg/runtime/engines/singularity/process_linux.go]: Ooops, I was clearly not paying enough attention, thanks for catching this. Fixed.

**Review by cclerget — APPROVED** [review]

**Pull request merged**